contracts/src/systems/config/hyperstructure_config.cairo

```diff
@@ -0,0 +1,61 @@
+#[system]
+mod DefineHyperStructure {
+    use eternum::alias::ID;
+    use eternum::components::hyperstructure::{HyperStructureResource, HyperStructure};
+    use eternum::components::position::{Position, Coord};
+
+    use dojo::world::Context;
+
+    use core::poseidon::poseidon_hash_span;
+    use core::traits::Into;
+    use core::array::{SpanTrait};
+
+    #[derive(Copy, Drop, Serde)]
+    struct RequiredResource {
+        resource_type: u8,
+        amount: usize
+    }
+
+    fn execute(ctx: Context, mut resources: Span<RequiredResource>, coord: Coord) -> ID {   
```

Comment:
maybe add a TODO to not forget to put that in ./set_config when we know what resources should be needed for each hyperstruture.

Reply:
would add hyperstrucutre_type here as well then